shared/reducers/waiting.js

```diff
@@ -12,11 +12,11 @@ function reducer(state: Constants.State = initialState, action: Waiting.Actions)
     }
     case Waiting.decrementWaiting: {
       const {payload: {key}} = action
-      return initialState.update(key, 1, n => n - 1)
+      return state.update(key, 1, n => n - 1)
```

Comment:
@MarcoPolo this change isn't used by this PR, but it seems like the existing code was incorrect -- using `initialState` rather than `state` for the offsets turned this into something more like a mutex than a semaphore.

(i.e. calling incrementWaiting twice left the count at 1, not 2.)

What do you think?